packages/menu/src/useMenu.ts

```diff
@@ -375,16 +375,52 @@ export const useMenu = <T extends HTMLElement = HTMLElement, M extends HTMLEleme
     [closeMenu, returnFocusToTrigger]
   );
 
-  const handleMenuBlur = useCallback(
-    (event: MouseEvent) => {
-      const path = event.composedPath();
+  /**
+   * 1. Determine if the next element receiving focus is focusable
+   *    (event.relatedTarget is null when focus moves to non-focusable elements or body).
+   *
+   * 2. When an element loses focus (on blur), and focus moves to a non-focusable element
+   *    like <body>, `event.relatedTarget` should be `null`. However, due to a bug in jsdom
+   *    (prior to version 24.1.2), `relatedTarget` is incorrectly set to the `Document` node
+   *    (`nodeName === '#document'`).
+   *
+   *    Currently, `jest-environment-jsdom` (v29.7.0) uses jsdom@20.0.3, which still has this issue.
+   *    Until Jest updates its jsdom dependency, this workaround ensures accurate
+   *    testing of focus behavior.
+   *
+   *    @see https://github.com/jsdom/jsdom/pull/3767
+   *    @see https://github.com/jsdom/jsdom/releases/tag/24.1.2
+   *    @see https://github.com/jestjs/jest/blob/v29.7.0/packages/jest-environment-jsdom/package.json
+   *
+   * 3. Skip focus-return to trigger in these scenarios:
+   *    a. Focus is moving to another focusable element
+   *    b. Menu is closed and focus would naturally go to body
+   */
+  const handleBlur = useCallback(
+    (event: React.FocusEvent) => {
+      const win = environment || window;
 
-      if (!path.includes(menuRef.current!) && !path.includes(triggerRef.current!)) {
-        returnFocusToTrigger();
-        closeMenu(StateChangeTypes.MenuBlur);
-      }
+      setTimeout(() => {
+        // Timeout is required to ensure blur is handled after focus
+        const activeElement = win.document.activeElement;
+        const isMenuOrTriggerFocused =
+          menuRef.current?.contains(activeElement) || triggerRef.current?.contains(activeElement);
+
+        if (!isMenuOrTriggerFocused) {
+          const nextElementIsFocusable =
+            !!event.relatedTarget /* [1] */ &&
+            event.relatedTarget?.nodeName !== '#document'; /* [2] */
```

Comment:
On our [Gatsby site](https://67ef78de3b212e875b90575a--zendeskgarden.netlify.app/), when a menu dropdown is open and you click outside, focus shifts to a `<div>` with `tabIndex="-1"` instead of the trigger. Should I handle that case?

![Screenshot 2025-04-03 at 4 53 53 PM](https://github.com/user-attachments/assets/bc49158f-03f1-4689-94f6-ab9f55b63cbb)

Reply:
Hmm, I doubt it, because the updated behavior is so much better than current https://garden.zendesk.com/components/menu – which jumps to the currently expanded menu when you trigger a menu further down the page.